llvm/lib/Target/RISCV/RISCVGatherScatterLowering.cpp

```diff
@@ -241,7 +250,7 @@ bool RISCVGatherScatterLowering::matchStridedRecurrence(Value *Index, Loop *L,
     return false;
   case Instruction::Or:
     // We need to be able to treat Or as Add.
-    if (!haveNoCommonBitsSet(BO->getOperand(0), BO->getOperand(1), *DL))
+    if (!cast<PossiblyDisjointInst>(BO)->isDisjoint())
```

Comment:
Very minor, but would you mind landing this as two patches.

Patch 1 - Add the disjoint or handling, but leave the haveNoCommonBitsSet case.
Patch 2 - Drop the haveNoCommonBitsSet

If dropping it exposes some missing inference case, I want the second to be easily revertable without taking out the rest of your change.